### ISSUE TITLE
[BUG] Indexes are not generated 

### ISSUE THREAD

**Issue by miroslavzeman** [jeffcampbellmakesgames/Entitas-Redux#53]
**Describe the bug**
Attributes `[EntityIndex]` and `[PrimaryEntityIndex]` do not generate additional code for getting entities from context.

**Unity Version:**
Unity 2020.3.0f1

**To Reproduce**
1. Create new Unity 2020.3.0f1 project
2. Update `manifest.json` with:
```
    "com.jeffcampbellmakesgames.genesis": "https://github.com/jeffcampbellmakesgames/genesis.git#release/stable",
    "com.jeffcampbellmakesgames.entitasredux": "https://github.com/jeffcampbellmakesgames/entitas-redux.git#releases/stable"
```
3. Setup genesis based on documentation -> click on `Update Genesis CLI` and `Install or Update All Plugins`
4. Create `GenesisSettings` and click on Auto Import
5. Create a class with content for testing the attributes:

```
using JCMG.EntitasRedux;

[Game]
public class PrimaryComponent : IComponent
{
    [PrimaryEntityIndex]
    public int value;
}

[Game]
public class SecondaryComponent : IComponent
{
    [EntityIndex]
    public int value;
}

public class Controller
{
    public Controller()
    {
        var context = new Contexts();
        // should contain something like: context.Game.GetEntityWithPrimary
        // should contain something like: context.Game.GetEntitiesWithSecondary
    }
}
```

6. Generate code via `Generate` button in `GenesisSettings` or shortcut `CTRL+SHIFT+G`

**Expected behavior**
Methods for getting data by indexes should be present in `Game` context, but they are not. It's not possible to use methods like `context.Game.GetEntityWithPrimary()` etc.

**Screenshots**

Genesis settings: 
![image](https://user-images.githubusercontent.com/10854425/123637625-86f8c180-d81e-11eb-8048-b9015f3f9b56.png)

Genesis setup in project settings:
![image](https://user-images.githubusercontent.com/10854425/123637746-a8f24400-d81e-11eb-89ca-5d553ae954d7.png)

**Additional context**
Win 10, using Mono .NET 4.x in Unity player settings

**Comment by jeffcampbellmakesgames** [comment]
I was able to track down and reproduce this bug locally. I was able to inspect further and have a feeling I know what the error is. I should have some more info in a day or two as to a fix.

**Comment by jeffcampbellmakesgames** [comment]
Looking at the implementation of `EntityIndexDataProvider.cs` [here](https://github.com/jeffcampbellmakesgames/Entitas-Redux/blob/29e6a0071a2b7c1442201e9a9968da9cf21459e8/Plugins/EntitasRedux.Core.Plugins/EntityIndex/DataProviders/EntityIndexDataProvider.cs#L84), it appears to be checking for any members of a component decorated with inherited attribute types of `AbstractEntityIndexAttribute`. However the extension/helper methods for HasAttribute and GetAttributes do not check base classes of attributes. As a result it would never find the Primary or Entity index attributes which derive from `AbstractEntityIndexAttribute`.

To resolve this I am approaching this as a few steps:

- [x] Add overloads or additions to the HasAttributes and GetAttributes methods in Genesis to offer support for checking base classes of attributes. Unit tests will also be created to ensure this functionality remains working.

- [x] After that Genesis functionality has been added and pushed, I will update this data provider accordingly to use it which should resolve being able to discover those derived attribute types again and generate the proper entity index code.

- [ ] One of the reasons this never came up earlier is that unit tests use manually created entity index code and there are no examples of using entity indexes in the development project. I will add common examples of both indexes to ensure if this kind of issue comes up in the future that it is discoverable when regenerating code.

**Comment by jeffcampbellmakesgames** [comment]
I've completed the first step of this which was to update genesis with better functionality for checking inherited attributes. Released this morning as a part of Genesis `v2.3.1`. I'll be taking a look at the data provider either tomorrow or this weekend.

**Comment by jeffcampbellmakesgames** [comment]
Just wanted to share an update on this; I have a fix prepared and am tracking down a remaining issue. I should have a new version containing this fix available mid-week.

**Comment by jeffcampbellmakesgames** [comment]
I've fixed this issue and merged to develop, the fix for this should soon be available in Entitas-Redux `v2.1.0`. If you could retest on this new version and let me know if this resolves your issue, I'd appreciate it.

**Comment by miroslavzeman** [comment]
Working like a charm! 💯 Thank you! 👍
